### DISCUSSION TITLE
About Next.js 13 removal of devDependencies

### DISCUSSION THREAD

**Discussion by eedemir** [vercel/vercel#9013] [Help] [answered]
Hello, I created Next.js project with create-next-app and I noticed something in package.json. 

```
"dependencies": {
    "@types/node": "18.11.10",
    "@types/react": "18.0.26",
    "@types/react-dom": "18.0.9",
    "eslint": "8.29.0",
    "eslint-config-next": "13.0.6",
    "next": "13.0.6",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "typescript": "4.9.3"
  }
```

AFAIK, types, eslint and typescript should be in devDependencies instead of dependencies.

I would like to know if there is a special reason for removing devDependencies at commit edd477e602834ec73201a867f93a9a1bb777e903 and adding those removed devDependencies in dependencies at commit 832ba3fe2332365779da6a079084bde6f46da124. Or is it just a bug?

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @eedemir! I don't have the answer on this one yet, but I'm sharing it with the Next.js team and I'll let you know when I learn more.

**Reply by amyegan** [reply]
Thanks for waiting! It seems that `devDependencies` may not be a reliable way to separate packages. So this change was intentional. 

The team recommends using [Output File Tracing](https://nextjs.org/docs/advanced-features/output-file-tracing) as a more reliable way to reduce deployment size. It will automaticaly trace the dependencies of each page to determine which files are needed for deploying your application. You can learn more about it at [nextjs.org/docs/advanced-features/output-file-tracing](https://nextjs.org/docs/advanced-features/output-file-tracing)

**Reply by kidroca** [reply]
I can see how this can help newcomers to npm avoid some pitfalls wondering whether to install a package as dependency or dev dependency.

Is it OK to manually install / move some dev only packages under `devDependencies` or is it expected to install everything under `dependencies` from now on?

Libraries like `prettier` would document their install steps as `npm install --save-dev ...` (same for `tailwind`), would it be a problem if users follow such instructions and install packages under `devDependencies`? 

I would assume packages for commit hooks like `husky` should not be installed under `dependencies`

**Comment by evalarumbe** [comment]
> Is it OK to manually install / move some dev only packages under devDependencies or is it expected to install everything under dependencies from now on?

I had the same question and this [video](https://www.youtube.com/watch?v=GlboxtkbT2s&ab_channel=CodinginFlow) helped me. To summarize: Next.js will automatically detect which dependencies aren't necessary to deliver in the build (that's what [Output File Tracing](https://nextjs.org/docs/advanced-features/output-file-tracing) is about), so manually marking deps as `devDependencies` is not necessary. It won't hurt if you still want to for some reason though.
